### PR TITLE
First steps with Typescript

### PR DESCRIPTION
**Pull request by martaqh**
This modest PR contains:
- Typescript installation and setup
- a few components rewritten from Options API to Composition API with script setup and Typescript
- typing of props, emit payload, functions arguments
- definition of an interface
- a few minor side fixes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,8 @@
     "test": "vitest"
   },
   "dependencies": {
+    "@rushstack/eslint-patch": "^1.2.0",
+    "@vue/eslint-config-typescript": "^11.0.2",
```
Comment (szymonlicau) [inline package.json]: This one can be moved to dev dependencies

File: .eslintrc.cjs [hunk 2]
```diff
@@ -1,9 +1,11 @@
 /* eslint-env node */
+require("@rushstack/eslint-patch/modern-module-resolution");
```
Comment (szymonlicau) [inline .eslintrc.cjs]: What's this one for? 🤔

File: src/components/pages/ChooseTicketsPage.vue [hunk 3]
```diff
@@ -1,10 +1,8 @@
...
-const props = defineProps({
-  seatsSelected: {
-    type: Array,
-    required: true
-  }
-})
+<script lang="ts" setup>
+
+const props = defineProps<{
+  seatsSelected: Array<string>
```
Comment (szymonlicau) [inline src/components/pages/ChooseTicketsPage.vue]: You can also use the shorthand if you like it more 😄
The same goes for all the other places.

```suggestion
  seatsSelected: string[]
```

File: src/components/pages/SingleMoviePage.vue [hunk 4]
```diff
@@ -1,54 +1,36 @@
...
-    },
-  },
-  mounted() {
-    this.getMovieDetails()
+const props = defineProps<{
+  movieId: string
+}>();
+
+const isRequestFinished = ref(false)
+const movieData = ref<Movie>(null)
```
Comment (szymonlicau) [inline src/components/pages/SingleMoviePage.vue]:
```suggestion
const movieData = ref<Movie | null>(null)
```

**Review by szymonlicau — CHANGES_REQUESTED** [review]

File: .eslintrc.cjs (hunk 2 above)
Comment (martaqh) [reply .eslintrc.cjs]: No idea 😉 I have just used `.eslintrc.cjs` from Tomek's demo repo to be sure TS works fine in my cinema. Seems I can remove this one and nothing explodes 😉

Comment (szymonlicau) [reply .eslintrc.cjs]: You definitely can 😄

File: src/components/pages/ChooseTicketsPage.vue (hunk 3 above)
Comment (martaqh) [reply src/components/pages/ChooseTicketsPage.vue]: I actually have tried both syntaxes on purpose -- as an exercise 😉 I know it looks messy to have both in the code -- I will change it for the shorthand everywhere.

Comment (szymonlicau) [reply src/components/pages/ChooseTicketsPage.vue]: Oh, it makes sense then 😄 Great!

**Review by szymonlicau — APPROVED** [review]

**Pull request merged**